### DISCUSSION TITLE
Panic when I trying to use gocron

### DISCUSSION THREAD

**Discussion by Enferum** [pocketbase/pocketbase#894] [Q&A] [answered]
Hi, could you help me to understand what exactly I'm doing wrong?
In my case, I had to add a scheduler that will check the Collection and do some staff.
Example:
```
func main() {
	app := pocketbase.New()

        s := gocron.NewScheduler(time.UTC)
	s.Every(5).Second().Do(func() {
		log.Println(app.Dao().FindCollectionByNameOrId("event"))
	})
	s.StartAsync()

	if err := app.Start(); err != nil {
		log.Fatal(err)
	}
}
```
run command - go run main.go serve
![image](https://user-images.githubusercontent.com/19825000/199449844-6262b83b-420a-4a14-94ce-cf7fe2183a87.png)

**Comment by ganigeorgiev** [comment, marked as answer]
This is because the database connection and all other related fields are not initialized until `app.Start()` (or more precisely until the internal `app.Bootstrap()` call; this is [documented in the godoc defintiion](https://pkg.go.dev/github.com/pocketbase/pocketbase#New) but I'll consider stating it more clear).

To fix the issue you can move your gocron initialization in the `OnBeforeServe` hook:

```go
app.OnBeforeServe().Add(func(e *core.ServeEvent) error {
    s := gocron.NewScheduler(time.UTC)
    s.Every(5).Second().Do(func() {
        c, err := app.Dao().FindCollectionByNameOrId("event")
        log.Println(c, err)
    })
    s.StartAsync()

    return nil
})
```

**Reply by Enferum** [reply]
Many thanks for your help.
